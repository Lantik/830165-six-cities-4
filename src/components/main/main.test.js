import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Main from './main.jsx';

const mock = {
  offers: [
    {
      coordinates: [52.3809553943508, 4.939309666406198]
    },
    {
      coordinates: [52.3809553943508, 4.939309666406198]
    },
    {
      coordinates: [52.3809553943508, 4.939309666406198]
    }]
};

it(`<Main/> rendered correctly`, () => {
  const {offers} = mock;
  const renderer = new ShallowRenderer();

  renderer.render(<Main offers={offers} onOfferHeaderClick={() => {}}/>);
  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});
