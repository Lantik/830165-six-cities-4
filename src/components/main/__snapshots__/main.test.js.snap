// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> rendered correcyly without offers in active city 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <CityList
          activeCity="London"
          cities={
            Array [
              "Amsterdam",
              "Capenhagen",
            ]
          }
          onCityTitleClick={[Function]}
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property availbale at the moment in 
              London
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;

exports[`<Main/> rendered correctly with offers in active city 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <CityList
          activeCity="Amsterdam"
          cities={
            Array [
              "Amsterdam",
              "Capenhagen",
            ]
          }
          onCityTitleClick={[Function]}
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2
             places to stay in 
            Amsterdam
          </b>
          <SortOptions
            onSortOptionClick={[Function]}
            sortType="DEFAULT"
          />
          <CitiesOfferList
            offers={
              Array [
                Object {
                  "city": "Amsterdam",
                  "coordinates": Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                },
                Object {
                  "city": "Amsterdam",
                  "coordinates": Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                },
              ]
            }
            onOfferCardMouseEnter={[Function]}
            onOfferHeaderClick={[Function]}
          />
        </section>
        <div
          className="cities__right-section"
        >
          <WithMap
            className="cities__map"
            coords={
              Array [
                Array [
                  52.3809553943508,
                  4.939309666406198,
                ],
                Array [
                  52.3809553943508,
                  4.939309666406198,
                ],
              ]
            }
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
